### PR TITLE
Limit the docker memory to {HOST_MEM-2G}, preventing OOMs

### PR DESCRIPTION
**Pull request by Felixoid**
### Changelog category (leave one):
- CI Fix or Improvement (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Attempt to prevent the runner's sudden death on OOMs.

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Run only fuzzers related jobs (libFuzzer fuzzers, AST fuzzers, etc.)
- [ ]  Exclude: AST fuzzers
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

**Comment by robot-ch-test-poll4** [comment]
*This is an automated comment for commit 726054d161a90ee94a3daa8a034a3398cf5d3cd6 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/72835/726054d161a90ee94a3daa8a034a3398cf5d3cd6/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Fast test</td><td>Normally this is the first check that is ran for a PR. It builds ClickHouse and runs most of <a href="https://clickhouse.com/docs/en/development/tests#functional-tests">stateless functional tests</a>, omitting some. If it fails, further checks are not started until it is fixed. Look at the report to see which tests fail, then reproduce the failure locally as described <a href="https://clickhouse.com/docs/en/development/tests#functional-test-locally">here</a></td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/b51df5655cf63eec05ba8703330a267f3a57ca34/fast_test.html">✅ success</a></td></tr>
<tr><td>Unit tests</td><td>Runs the unit tests for different release types</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/b51df5655cf63eec05ba8703330a267f3a57ca34/unit_tests__release_.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

**Comment by Felixoid** [comment]
https://github.com/ClickHouse/ClickHouse/actions/runs/12179546344/job/33972591499?pr=72835#step:8:4554 it worked out

File: tests/integration/runner [hunk 1]
```diff
@@ -276,6 +289,14 @@ if __name__ == "__main__":
         help="Set network driver for runnner container (defaults to `host`)",
     )
 
+    parser.add_argument(
+        "--memory",
+        # Default is host memory - 2GiB
+        default=physical_memory() - 2 * 1024**3,
```
Comment (maxknv) [inline tests/integration/runner:295]: would be nice to validate the result value I think, check that it is >0 or >PHYSICAL_MEMORY/2

**Comment by maxknv** [comment]
LGTM

File: tests/integration/runner (hunk 1 above)
Comment (Felixoid) [reply tests/integration/runner:295]: It's legal `docker run --memory` argument. I've checked

> -m , --memory=number[*S]
>   Memory limit; S is an optional suffix which can be one of b, k, m, or g.
>
> Allows  you  to  constrain the memory available to a container. If the host supports swap memory, then the -m memory setting can be larger than physical RAM. If a limit of 0 is specified (not using -m), the container's memory is not limited. The actual limit may be rounded up to a multiple of the operating system's page size (the value would be very large, that's millions of trillions).

**Pull request merged**